front-end/src/pages/newActivity.js

```diff
@@ -31,15 +31,18 @@ class NewActivity extends Component {
   }
 
   handleSubmit(){
-    this.props.dispatch(createNewActivity(this.props.apiUrl, this.state.form))
+    this.props.onSubmit(this.state.form)
+    this.setState({form:{ name: '', description:'', address: '', latitude:'', longitude:''}})
+    document.getElementById("submitMessage").innerText = "Activity Submitted"
+    setTimeout(function(){document.getElementById("submitMessage").innerText = ""}, 2000)
```

Comment:
why a timeout here?

Reply:
To remove the "Activity Submitted" text after it displays for 2 seconds.